torch/ao/quantization/backend_config/backend_config.py

```diff
@@ -63,32 +72,84 @@ class ObservationType(Enum):
     example: torch.cat, maxpool
     """
 
+
+@dataclass
+class DTypeWithConstraints:
+    """
+    Config for specifying additional constraints for a given dtype, such as quantization value
+    ranges and scale value ranges, to be used in :class:`~torch.ao.quantization.backend_config.DTypeConfig`.
+    """
+    dtype: torch.dtype
+    quant_min: Optional[float] = None
+    quant_max: Optional[float] = None
+    scale_min: Optional[float] = None
+    scale_max: Optional[float] = None
+
+
 @dataclass
 class DTypeConfig:
     """
     Config for the set of supported input/output activation, weight, and bias data types for the
     patterns defined in :class:`~torch.ao.quantization.backend_config.BackendConfig`.
     """
-    input_dtype: Optional[torch.dtype] = None
-    output_dtype: Optional[torch.dtype] = None
-    weight_dtype: Optional[torch.dtype] = None
+    input_dtype: Union[torch.dtype, DTypeWithConstraints, None] = None
+    output_dtype: Union[torch.dtype, DTypeWithConstraints, None] = None
+    weight_dtype: Union[torch.dtype, DTypeWithConstraints, None] = None
     bias_dtype: Optional[torch.dtype] = None
     is_dynamic: Optional[bool] = None
 
+    def get_input_dtype(self) -> Optional[torch.dtype]:
+        """
+        Return the input dtype as a `torch.dtype`.
+        """
+        if isinstance(self.input_dtype, DTypeWithConstraints):
+            return self.input_dtype.dtype
+        else:
+            return self.input_dtype
```

Comment:
maybe we can define a more general helper that's given a union and return the dtype, and use that helper function, instead of repeat this code multiple times

Reply:
I think if we always store `DTypeWithConstraints` then we don't need to do this